### ISSUE TITLE
[HOLD for payment 2024-08-09] [$250] Xero - Xero action modal appears below the button and overlaps with the composer

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45607]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.8-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/44170
## Action Performed:
Precondition:
- Workspace is connected to Xero
- Delayed submission is enabled in Workflows
1. Go to staging.new.expensify.com
2. Go to workspace chat
3. Submit an expense
4. In the main chat, click on the pay elsewhere dropdown.the composer.
6. Pay elsewhere the report
7. Click on the Xero action dropdown

## Expected Result:
Xero action modal will appear above the button and does not overlap with the composer

## Actual Result:
Xero action modal appears below the button and overlaps with the composer

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/c20d38e5-f41b-46b0-8c69-384acca67b66

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019da83e276a5e9487</li>
        <li>Upwork Job ID: 1813781422967508952</li>
        <li>Last Price Increase: 2024-07-18</li>
<li>Automatic offers: </li>
<ul>
<li>nyomanjyotisa | Contributor | 103231671</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 2

**Comment by trjExpensify** [comment]
@arosiclair do you agree this isn't a blocker? 

I agree the popover should be in the same place as the pay selector options, so let's fix it. Also, we should be using circle Xero avatars in the popover, not squares. Just like QBO:

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/214dc8d5-e021-4b78-9934-c168aa2e737f">

**Comment by trjExpensify** [comment]
Certainly not a web blocker, removing that label minimally now.

**Comment by arosiclair** [comment]
> @arosiclair do you agree this isn't a blocker?

Yeah this looks pretty minor to me I agree it shouldn't be a blocker.

**Comment by arosiclair** [comment]
Proposal looks okay though having the dropdown always open _upward_ feels off to me tbh. Also you could argue that the dropdown covering the preview is just as bad (or even worse) than covering the composer.

Played around with Slack and it looks like it only opens the context menu upwards when there isn't enough space below regardless of the composer:

| Down | Up |
|--------|--------|
| ![Screenshot 2024-07-22 at 11 38 51 AM](https://github.com/user-attachments/assets/98f130c2-9219-40ef-8059-eff48f6d2845) | ![Screenshot 2024-07-22 at 11 39 26 AM](https://github.com/user-attachments/assets/04733467-6946-468f-9efd-0b1b9ac048cb) | 

So I'm starting to think the dropdown covering the composer isn't really a problem and we really should just be worried about it not fitting on screen (which I think isn't an issue yet). What do you think @trjExpensify?

**Comment by trjExpensify** [comment]
>So I'm starting to think the dropdown covering the composer isn't really a problem and we really should just be worried about it not fitting on screen (which I think isn't an issue yet). What do you think @trjExpensify?

I think it should be consistent with the `Pay` button in the same preview component, so I'd say stick with how this menu opens today for that (which is upwards). If we want to make a change, we should do it overall, and I don't think it's worth the effort to revisit this one now. So I'd just fix the consistency issue with the export menu. 👍

**Comment by arosiclair** [comment]
Fair enough. We can look at it another time 👍

**Comment by trjExpensify** [comment]
@nyomanjyotisa look forward to a PR for this. Can you make sure to change those Xero icons to circles instead of squares to match QBO in the PR? Thanks! 

<img width="811" alt="image" src="https://github.com/user-attachments/assets/cbb0f68f-a5f1-49a2-86e7-b999e3c2d606">

<img width="805" alt="image" src="https://github.com/user-attachments/assets/cc7abc36-d803-4f43-8a98-04dc6a8c9c5d">

**Comment by trjExpensify** [comment]
Cool, thanks! CC: @dannymcclain for vis just in case that intentionally changed somewhere.

**Comment by dannymcclain** [comment]
Nope, they're supposed to be circles. We just like to upload the assets to the repo as squares and then do the rounding with code (in case we ever want to change them from circles to rounded squares or adjust the border radius or something).

**Comment by trjExpensify** [comment]
Perf, thanks for confirming!

**Comment by trjExpensify** [comment]
I think we're splitting hairs on requesting a bounty increase for this issue. The position of the popover was wrong appearing over the composer, and we needed to put it back to where it should be. Switching `XeroSquare` for `XeroCircle` is also so incredibly minor. Overall, $250 is already a generous bounty for a +29 line PR. 

<img width="191" alt="image" src="https://github.com/user-attachments/assets/187ee5b2-5cb3-40cc-a864-a4d559323ad0">

**Comment by kadiealexander** [comment]
Payouts due:

- [x] Contributor: $250 @nyomanjyotisa [automatic offer](https://www.upwork.com/nx/wm/offer/103231671)
- [ ] Contributor+: $250 @aimane-chnaif via NewDot

Upwork job is [here](upwork.com/jobs/~019da83e276a5e9487).

**Comment by kadiealexander** [comment]
@aimane-chnaif don't forget the checklist!

**Comment by kadiealexander** [comment]
Thanks @aimane-chnaif! Please apply via NewDot (summary [here](https://github.com/Expensify/App/issues/45607#issuecomment-2277048362))
